RNA-seq/00c-tximeta_other_species.Rmd

```diff
@@ -0,0 +1,213 @@
+
+![](diagrams/rna-seq_5.png)
+
+For more information about `tximeta`, see [this excellent vignette](https://www.bioconductor.org/packages/devel/bioc/vignettes/tximeta/inst/doc/tximeta.html) from Love _et al_.
+
+## Libraries and functions
+
+```{r library}
+# Load magrittr for the pipe
+library(magrittr)
+
+# Load the tximeta package
+library(tximeta)
+
+# Load the SummarizedExperiment package
+library(SummarizedExperiment)
+```
+
+## Directories and files
+
+In this instance, we will need to provide `tximeta` with the salmon index for the species we are using. 
+We have prepared this for zebrafish along with other species which you can find more information about [on GitHub](https://github.com/AlexsLemonade/training-txome-prep).
+
+```{r directories}
+# Set up file path to the salmon index for the species we are using
+index_dir <- file.path("index", "Danio_rerio", "short_index")
+```
+
+These directories and files are for pointing to the `quant.sf` files. 
+
+```{r}
+# directory where the data are located
+data_dir <- file.path("data", "zebrafish-cortisol")
+
+# directory where the quant files are located, each sample is its own
+# directory
+quant_dir <- file.path(data_dir, "salmon_quant")
+
+# create a directory to hold the tximport results if it doesn't exist yet
+txi_dir <- file.path(data_dir, "tximport")
+if (!dir.exists(txi_dir)) {
+  dir.create(txi_dir, recursive = TRUE)
+}
+```
+
+```{r input-names}
+# the quant files themselves
+sf_files <- list.files(quant_dir, recursive = TRUE, full.names = TRUE,
+                       pattern = "quant.sf")
+```
+
+```{r metadata-file}
+# sample metadata file
+meta_file <- file.path(data_dir, "zebrafish-cortisol_metadata.tsv")
+```
+
+**Output**
+
+```{r output-names}
+# Name the output zebrafish-cortisol_tximport.RDS and use the directory created 
+# above as the rest of the path
+txi_out_file <- file.path(txi_dir, "zebrafish-cortisol_tximeta.RDS")
+```
+
+## Set up reference data for a non-human, non-mouse species 
+
+Here we will set this up for zebrafish, _Danio rerio_.
+You can look at Ensembl's ftp server to see the full list of species and fasta file options: http://ftp.ensembl.org/pub/release-98/fasta/
+
+If you are using a species other than zebrafish, click through the file paths to find the file you want to use for annotation. 
+
+We will store the `base_url` where all the species FASTA files are located and then the second part of the url is species-specific which you may have to replace. 
+
+```{r}
+fasta_base_url <- "ftp://ftp.ensembl.org/pub/release-98/fasta/"
+
+fasta_files <- c(
+  paste0(
+    fasta_base_url, 
+    # Replace the file path below to the species specific fasta cDNA file
+    "danio_rerio/cdna/Danio_rerio.GRCz11.cdna.all.fa.gz" 
+    ),
+  paste0(
+    fasta_base_url, 
+    # Replace the non-coding RNA fasta file path so that non-coding genes are also annotated
+    "danio_rerio/ncrna/Danio_rerio.GRCz11.ncrna.fa.gz" 
```

Comment:
Are these in our index? I don't think so, but I don't know for sure.

Reply:
They are not. It does not appear including them hurt anything but probably this information in this file doesn't actually go anywhere since its not included in our index in the first place.

Reply:
I dropped the ncRNA file because I don't think it was doing anything besides adding an extra step.